_posts/2018-10-10-ecto-multi.md

```diff
@@ -0,0 +1,124 @@
+---
+author: Svilen Gospodinov
+author_link: https://github.com/svileng
+categories: post
+date:   2018-10-10
+layout: post
+title:  A brief guide to Ecto.Multi
+excerpt: Learn how to compose and execute batches of queries using Ecto.Multi.
+---
+
+Ecto.Multi is a set of utilities aimed at composing and executing atomic operations, usually (but not always, as you’ll see soon) performed against the database. Furthermore, it handles rollbacks, provides results on either success or error, flattens-out nested code and saves multiple round trips to the database.
+
+If you find yourself running and managing many database queries (and other operations), then keep reading and you may find some useful tools to add your Elixir/Ecto toolbox.
+
+## Creating a Multi
+Everything starts with a `%Multi{}` struct. Regardless of what you’re doing, you always have to provide a new or an existing Multi to most of the functions, which you can easily create by calling `new()`:
+
+```elixir
+iex> Ecto.Multi.new()
+%Ecto.Multi{names: #MapSet<[]>, operations: []}
+```
+
+## Executing Multi operations
+To run a Multi, you have to hand it over to `Repo.transaction/1`:
+
+```elixir
+iex> Ecto.Multi.new() |> Repo.transaction()
+{:ok, %{}}
+```
+Clearly we just ran an empty Multi, which was obviously successful since nothing was performed (and nothing returned in the second element of the {:ok, return} tuple. To make Multis useful, you need to add operations to it.
+
+Next, we're going to cover the most common operations you may end up doing.
+
+## Working with individual changesets
+When working with multiple `%Ecto.Changeset{}`s, usually you will call `Repo.insert/1` / `update/1` etc multiple times to run the operations. Switching to `Ecto.Multi` is as easy as replacing `Repo.update/1` with its equivalent `Ecto.Multi.update/3`, for example.
+
+Assuming you already have a `team_changeset`, `user_changeset` and `foo_changeset` created beforehand, this would look like so:
+
+```elixir
+Ecto.Multi.new()
+|> Ecto.Multi.insert(:team, team_changeset)
+|> Ecto.Multi.update(:user, user_changeset)
+|> Ecto.Multi.delete(:foo, foo_changeset)
+|> Repo.transaction()
+```
+The atoms used — `:user` ,`:team` and `:foo` — are up to you. You can pass anything (also you can use a string, instead of an atom) as long as it’s unique value for the current Multi.
```

Comment:
You probably want to add an `a`, making it `as long as it's a unique value for the current Multi.`